sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -88,23 +96,23 @@ case class CreateViewCommand(
     qe.assertAnalyzed()
     val analyzedPlan = qe.analyzed
 
-    if (tableDesc.schema != Nil && tableDesc.schema.length != analyzedPlan.output.length) {
+    if (userSpecifiedColumns.nonEmpty &&
```

Comment:
We allow users to create a table containing zero column, right? See my previous PR: https://github.com/apache/spark/pull/14007